src/scripts/main.js

```diff
@@ -1,3 +1,135 @@
 'use strict';
 
-// write code here
+// sort function
+const tHeadTags = document.querySelectorAll('th'); // Вибираємо всі заголовки
+const tbody = document.querySelector('tbody'); // Тіло таблиці
+const rows = tbody.querySelectorAll('tr'); // Всі рядки
+
+const sortMode = {}; // Об'єкт для відстеження стану сортування
+
+tHeadTags.forEach((tag, thIndex) => {
+  // Початковий стан сортування — за зростанням
+  sortMode[thIndex] = 'asc';
+
+  tag.addEventListener('click', () => {
+    const direction = sortMode[thIndex];
+
+    if (direction === 'asc') {
+      sortTable(thIndex, 'asc');
+      sortMode[thIndex] = 'desc'; // Змінюємо стан
+    } else {
+      sortTable(thIndex, 'desc');
+      sortMode[thIndex] = 'asc'; // Змінюємо стан
+    }
+  });
+});
+
+function sortTable(indx, mode) {
+  const table = document.querySelector('table');
+  const content = Array.from(table.rows).slice(1, -1); // Пропускаємо заголовки
+
+  content.sort((a, b) => {
+    const cellA = a.cells[indx].innerText.trim();
+    const cellB = b.cells[indx].innerText.trim();
+
+    if (mode === 'asc') {
+      return cellA > cellB ? 1 : -1;
+    } else {
+      return cellA < cellB ? 1 : -1;
```

Comment:
When comparing `cellA` and `cellB`, ensure they are of the same type. If these are numbers, convert them using `parseFloat` or `parseInt` before comparison to avoid lexicographical sorting.